addOns/ascanrulesBeta/src/main/java/org/zaproxy/zap/extension/ascanrulesBeta/RelativePathConfusionScanRule.java

```diff
@@ -327,7 +327,9 @@ public void scan() {
                             // attribute)
                             if (tag.toUpperCase().equals("STYLE")) {
                                 // for the style tag, look at the entire body, not an attribute..
-                                String styleBody = tagInstance.data();
+                                // remove all " and ' for proper matching url('somefile.png')
+                                String styleBody =
+                                        tagInstance.data().replaceAll("'", "").replaceAll("\"", "");
```

Comment:
This can be done through the regex.

Reply:
The regex is `STYLE_URL_LOAD`.

Reply:
Ah yeah right, I tried to handle this by regex but after spending almost 1 hour I gave up... If you have some advice I will be grateful.

Reply:
Add `['\"]?` before the `https?`, if that does not help we should extract a method to handle this case since `STYLE_URL_LOAD` is used in more than this place.

Reply:
Also worth updating `shouldNotAlertIfAbsolutePathIsDetected` to have the absolute path also quoted (like it was done for the relative).

Reply:
@thc202 refactored code to use one method + added tests you mentioned